src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-cluster.h

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ *    Copyright (c) 2024-25 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-logic.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <clusters/WiFiNetworkDiagnostics/ClusterId.h>
+#include <platform/DiagnosticDataProvider.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class WiFiDiagnosticsServerCluster : public DefaultServerCluster
+{
+public:
+    WiFiDiagnosticsServerCluster(EndpointId endpointId, DeviceLayer::DiagnosticDataProvider & diagnosticProvider,
```

Comment:
if we know endpointid is always root (we hardcode some things), should we drop endpointID from here and use kRoot?

Reply:
Done

Reply:
We should have endpointId, the cluster can be part of any endpoint. https://github.com/project-chip/connectedhomeip/pull/39898#discussion_r2231192857